src/core/components/AuthenticateButton/index.js

```diff
@@ -8,27 +10,37 @@ import { logOutFromServer, startLoginUrl } from 'core/api';
 import translate from 'core/i18n/translate';
 import Button from 'ui/components/Button';
 import Icon from 'ui/components/Icon';
+import type { UrlFormatParams } from 'core/api';
+import type { ApiStateType } from 'core/reducers/api';
+import type { DispatchFn } from 'core/types/reduxTypes';
 
+type HandleLogInFn = (
+  location: UrlFormatParams, options?: {| _window: typeof window |}
+) => void;
+
+type HandleLogOutFn = ({| api: ApiStateType |}) => Promise<void>;
+
+type AuthenticateButtonProps = {|
+  api: ApiStateType,
```

Comment:
yeah, being able to look at a prop that expects an object and follow it to the type definition to know what the object looks like is a major win.